core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java

```diff
@@ -390,7 +390,12 @@ private RemoteTransaction getOrCreateRemoteTransaction(GlobalTransaction globalT
             if (topologyId < minTxTopologyId) {
                if (trace)
                   log.tracef("Changing minimum topology ID from %d to %d", minTxTopologyId, topologyId);
-               minTxTopologyId = topologyId;
+               minTopologyRecalculationLock.lock();
```

Comment:
Yes you're correct. I encountered a test failure shortly after issuing the PR anyway :confounded: